src/responder/kcm/kcmsrv_ops.c

```diff
@@ -1182,20 +1290,49 @@ kcm_op_get_cred_by_uuid_send(TALLOC_CTX *mem_ctx,
 {
     struct tevent_req *req = NULL;
     struct tevent_req *subreq = NULL;
-    struct kcm_op_common_state *state = NULL;
+    struct kcm_op_get_cred_by_uuid_state *state;
+    struct kcm_cred *crd;
     errno_t ret;
     const char *name;
 
-    req = tevent_req_create(mem_ctx, &state, struct kcm_op_common_state);
+    req = tevent_req_create(mem_ctx, &state,
+                            struct kcm_op_get_cred_by_uuid_state);
     if (req == NULL) {
         return NULL;
     }
-    state->op_ctx = op_ctx;
+    state->common.op_ctx = op_ctx;
 
     ret = sss_iobuf_read_stringz(op_ctx->input, &name);
     if (ret != EOK) {
         goto immediate;
     }
+
+    ret = sss_iobuf_read_len(state->common.op_ctx->input, UUID_BYTES,
+                             state->uuid);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_OP_FAILURE, "Cannot read input UUID [%d]: %s\n",
+              ret, sss_strerror(ret));
+        goto immediate;
+    }
+
+    if (op_ctx->conn_data->creds != NULL) {
+        crd = kcm_creds_lookup(op_ctx->conn_data->creds, state->uuid);
+        if (crd == NULL) {
```

Comment:
Shall not we try to update table here (i.e. `kcm_ccdb_getbyname_send()`)?

Reply:
Or call get-by-uuid without previous get-uuid-list in unexpected?

Reply:
This should not happen. I added a comment there.  The way how it is supposed to be used is to retrieve list of UUIDs in ccache (`GET_CRED_UUID_LIST`) then iterate over those with `GET_CRED_BY_UUID`. So if someone calls `GET_CRED_BY_UUID` without `GET_CRED_UUID_LIST` or if someone passed wrong UUID, it's not a reason to rebuild the table.

Reply:
Ok, thanks for clarification.